browser/ephemeral_storage/ephemeral_storage_browsertest.cc

```diff
@@ -449,3 +450,56 @@ IN_PROC_BROWSER_TEST_F(EphemeralStorageBrowserTest,
   EXPECT_EQ("", private_values.iframe_1.cookies);
   EXPECT_EQ("", private_values.iframe_2.cookies);
 }
+
+IN_PROC_BROWSER_TEST_F(EphemeralStorageBrowserTest,
+                       NavigationCookiesArePartitioned) {
+  AllowAllCookies();
+
+  GURL a_site_set_cookie_url = https_server_.GetURL(
+      "a.com", "/set-cookie?name=acom;path=/;SameSite=None;Secure");
+  GURL b_site_set_cookie_url = https_server_.GetURL(
+      "b.com", "/set-cookie?name=bcom;path=/;SameSite=None;Secure");
+
+  ui_test_utils::NavigateToURL(browser(), a_site_set_cookie_url);
+  ui_test_utils::NavigateToURL(browser(), b_site_set_cookie_url);
+  ui_test_utils::NavigateToURL(browser(), a_site_ephemeral_storage_url_);
+
+  std::string a_cookie =
```

Comment:
are we covering all the cases here? For instance what about verifying that 1st-party iframe does not use ephemeral storage with and without being nested in 3p domain?

Reply:
Are you referring to the following situation?
Top frame:
  * 3rd party iframe
      * 1st party iframe

Reply:
correct

Reply:
I've added a new browsertest for this functionality.

Reply:
I'm not sure that was the only one that is missing, but please check with @pes10k and add any others in a follow-up